backend/src/lib/services/friendlyurl-service.ts

```diff
@@ -43,12 +43,11 @@ function isValid(friendlyURL: string): boolean {
 
 function generateFriendlyURL(dashboardName: string): string {
   return dashboardName
-    .trim()
     .toLocaleLowerCase()
     .replace(/[!#$&'\(\)\*\+,\/:;=\?@\[\]]+/g, " ") // remove RFC-3986 reserved characters
+    .trim() // remove spaces from the beginning and the end.
     .replace(/\s+/g, "-") // replace spaces for dashes
-    .replace(/-+/g, "-") // convert consecutive dashes to singular dash
-    .replace(/^(-)+|(-)+$/g, ""); // remove dashes at the end and beginning
```

Comment:
what about dashes at the beginning or the end

```
"-test-name-"
```

Reply:
the stated test case was added to the method as well to the unit tests.